pkg/evaluators/identity/api_key.go

```diff
@@ -26,18 +28,21 @@ type APIKey struct {
 	Name           string              `yaml:"name"`
 	LabelSelectors k8s_labels.Selector `yaml:"labelSelectors"`
 	Namespace      string              `yaml:"namespace"`
+	KeySelectors   []string            `yaml:"keySelectors"`
 
+	// Map of API Key value to secret
 	secrets   map[string]k8s.Secret
 	mutex     sync.RWMutex
 	k8sClient k8s_client.Reader
 }
 
-func NewApiKeyIdentity(name string, labelSelectors k8s_labels.Selector, namespace string, authCred auth.AuthCredentials, k8sClient k8s_client.Reader, ctx context.Context) *APIKey {
+func NewApiKeyIdentity(name string, labelSelectors k8s_labels.Selector, namespace string, keySelectors []string, authCred auth.AuthCredentials, k8sClient k8s_client.Reader, ctx context.Context) *APIKey {
 	apiKey := &APIKey{
 		AuthCredentials: authCred,
 		Name:            name,
 		LabelSelectors:  labelSelectors,
 		Namespace:       namespace,
+		KeySelectors:    append(keySelectors, defaultAPIKeySelector),
```

Comment:
This means the user cannot invalidate `api_key` as a key then, no? Should we only `append` when `len(keySelectors) == 0` maybe?

Reply:
Yep. I guess so.